### PR TITLE
Feat: responsive designs adaptable to giant screens

### LINKED ISSUE #222: Responsive designs for giant screens

### PR DESCRIPTION
**Pull request by braianj**
Screenshot 2023-05-01 at 18 36 48
Screenshot 2023-05-01 at 18 37 01
Screenshot 2023-05-01 at 18 37 10
Screenshot 2023-05-01 at 18 37 47
Screenshot 2023-05-01 at 18 37 58
Screenshot 2023-05-01 at 18 38 15
Screenshot 2023-05-01 at 18 38 52
Screenshot 2023-05-01 at 18 39 01
Screenshot 2023-05-01 at 18 39 07
Screenshot 2023-05-01 at 18 39 14
Screenshot 2023-05-01 at 18 39 21
Screenshot 2023-05-01 at 18 39 29
Screenshot 2023-05-01 at 18 39 36
Screenshot 2023-05-01 at 18 39 43
Screenshot 2023-05-01 at 18 39 52

### CODE REVIEW

File: src/components/Layout/OverviewList.tsx [hunk 1]
```diff
@@ -10,6 +12,7 @@ import { HeaderMenu } from "decentraland-ui/dist/components/HeaderMenu/HeaderMen
 import Icon from "semantic-ui-react/dist/commonjs/elements/Icon"
 
 import { AggregatePlaceAttributes } from "../../entities/Place/types"
+import { useCardsByWidth } from "../../hooks/useCardsPerWindowSize"
```
Comment (2fd) [inline src/components/Layout/OverviewList.tsx:15]:
```suggestion
import useCardsByWidth from "../../hooks/useCardsByWidth"
```

following the convention: https://github.com/decentraland/decentraland-gatsby/blob/master/docs/guide/typescript.md#exports-named-vs-default-export

File: src/components/Layout/OverviewList.tsx [hunk 2]
```diff
@@ -40,6 +43,15 @@ export default React.memo(function OverviewList(props: OverviewListProps) {
     dataPlace,
   } = props
   const l = useFormatMessage()
+
+  const isBigScreen = useMediaQuery({ minWidth: 1800 })
```
Comment (2fd) [inline src/components/Layout/OverviewList.tsx:47]: 1800 looks too arbitrary, we should probably use `1280` or `1920` which are common screen sizes

image

also, we should also create a hook to contains that logic similar to the ones that already exists on [decentraland-ui](https://github.com/decentraland/ui/blob/master/src/components/Media/Media.ts#L6) and create an issue to move it out to this project, at least, to `decentraland-gatsby`

```suggestion
  const isBigDesktop = useBigDesktopMediaQuery()
```

File: src/components/Layout/OverviewList.tsx [hunk 3]
```diff
@@ -40,6 +43,15 @@ export default React.memo(function OverviewList(props: OverviewListProps) {
...
+
+  const cardsToShow = useCardsByWidth({
+    cardWidth: isBigScreen ? 310 : 260,
+    cardMargin: 14,
+    containerMargin: 48,
+  })
```
Comment (2fd) [inline src/components/Layout/OverviewList.tsx:53]: This looks too hardcoded 😅 , Isn't any other way to get these values directly from the components? That way, we can eventually create an standard component that handles all this nice behavior

File: src/components/Place/PlaceList/PlaceList.tsx [hunk 4]
```diff
@@ -36,13 +46,25 @@ export default React.memo(function PlaceList(props: PlaceListProps) {
...
+
+  const cardColumns = useMemo(() => cardsToShow.toString(), [cardsToShow])
+
   return (
     <div
       className={TokenList.join([
         "place-list__container",
         className && className,
       ])}
+      style={{ "--card-columns": cardColumns } as CustomCSSProperties}
```
Comment (2fd) [inline src/components/Place/PlaceList/PlaceList.tsx:67]: We should move this into a `useMemo`

File: src/components/Place/PlaceList/PlaceList.tsx [hunk 5]
```diff
@@ -36,13 +46,25 @@ export default React.memo(function PlaceList(props: PlaceListProps) {
...
+  const isTablet = useTabletAndBelowMediaQuery()
   const isMobile = useMobileMediaQuery()
+
+  const cardsToShow = useCardsByWidth({
+    cardWidth: isBigScreen ? 310 : 260,
+    cardMargin: 14,
+    containerMargin: isTablet ? 14 : 48,
+  })
+
+  const cardColumns = useMemo(() => cardsToShow.toString(), [cardsToShow])
```
Comment (2fd) [inline src/components/Place/PlaceList/PlaceList.tsx:59]:
```suggestion
  const style = useMemo(
     () => ({
         ...props.style,
         "--card-columns": cardColumns.toString()
      }),
      [cardsToShow, props.style]
)
```

File: src/hooks/useCardsPerWindowSize.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from "react"
+
+export function useCardsByWidth(options: {
```
Comment (2fd) [inline src/hooks/useCardsPerWindowSize.ts:3]:
```suggestion
export default function useCardsByWidth(options: {
```

File: src/hooks/useCardsPerWindowSize.ts
Comment (2fd) [inline src/hooks/useCardsPerWindowSize.ts:1]: the file should have the same name that the only function that is intended to export

File: src/pages/index.tsx [hunk 7]
```diff
@@ -51,8 +46,7 @@ export default function OverviewPage() {
     usePlaceListHightRated(overviewOptions)
   const [placeListMyFavorites, placeListMyFavoritesState] =
     usePlaceListMyFavorites(overviewOptions)
-  const [placeListPois, placeListPoisState] =
-    usePlaceListPois(overviewPoisOptions)
+  const [placeListPois, placeListPoisState] = usePlaceListPois()
```
Comment (2fd) [inline src/pages/index.tsx:49]:
```suggestion
  const [placeListPois, placeListPoisState] = usePlaceListPois(overviewOptions)
```

File: src/hooks/usePlaceListPois.ts [hunk 8]
```diff
@@ -5,26 +5,20 @@ import Places from "../api/Places"
 import { AggregatePlaceAttributes } from "../entities/Place/types"
 import { getPois } from "../modules/pois"
 
-export function usePlaceListPois(options?: {
-  limit: number
-  offset: number
-  positions?: string[]
-}) {
+export function usePlaceListPois() {
```
Comment (2fd) [inline src/hooks/usePlaceListPois.ts:8]: This should handle `offset` and `limit` the same way that `usePlaceListMyFavorites` does

**Review by 2fd — CHANGES_REQUESTED** [review]

File: src/components/Layout/OverviewList.tsx (hunk 3 above)
Comment (braianj) [reply src/components/Layout/OverviewList.tsx:53]: I removed the isBigDesktop and set a card fixed width in useCardsByWidth hook and also into the css.
Is this good enough or should I extract those numbers (cardWidth, cardMargin, containerMargin) dynamically using useRef?

Comment (2fd) [reply src/components/Layout/OverviewList.tsx:51]: We should detect it dynamically, but it is not necessarily urgent. We can merge it if there is and issue to fix it later.

**Review by 2fd — APPROVED** [review]
https://github.com/decentraland/places/issues/260

**Pull request merged**